data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/S3SourceConfig.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.source;
+
+import com.amazon.dataprepper.plugins.source.configuration.AWSAuthenticationOptions;
+import com.amazon.dataprepper.plugins.source.configuration.CodecOption;
+import com.amazon.dataprepper.plugins.source.configuration.CompressionOption;
+import com.amazon.dataprepper.plugins.source.configuration.NotificationTypeOption;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.NotNull;
+
+public class S3SourceConfig {
+
+    @JsonProperty("notification_type")
+    @NotNull
+    private NotificationTypeOption notificationType;
+
+    @JsonProperty("compression")
+    private CompressionOption compression = CompressionOption.NONE;
```

Comment:
Why is the `CompressionOption` set to NONE?

Reply:
This is the default value for compression. Logs received from the S3 can be uncompressed, for example `S3 access logs` and `ELB logs`. We don't have to perform any decompression for them.